src/com/mendix/widget/badge/components/Badge.ts

```diff
@@ -0,0 +1,152 @@
+    }
+
+    render() {
+        return this.setBadge();
+    }
+
+    private setBadge() {
+        if (this.props.badgeType === "button") {
+            return this.createBadgeButton();
+        } else if (this.props.badgeType === "label") {
+            return this.createBadgeLabel();
+        } else {
+            return this.createBadge();
+        }
+    }
+
+    private createBadge() {
+        return createElement("div",
+            {
+                className: classNames("widget-badge-display",
+                    { "widget-badge-link": !!this.props.badgeOnClick }
+                ),
+                onClick: () => this.handleOnClick(this.props.badgeOnClick)
+            },
+            DOM.span({ className: "widget-badge-text" }, this.props.label),
+            DOM.span({
+                className: classNames("widget-badge", "badge",
+                    { [`label-${this.props.style}`]: !!this.props.style }
+                )
+            }, this.props.badgeValue)
+        );
+    }
+
+    private createBadgeButton() {
+        return createElement("button",
+            {
+                className: classNames("widget-badge btn",
+                    { [`btn-${this.props.style}`]: !!this.props.style }
+                ),
+                disabled: this.props.disabled,
+                onClick: () => this.handleOnClick(this.props.badgeOnClick)
+            },
+            DOM.span({ className: "widget-badge-text" }, this.props.label),
+            DOM.span({ className: "badge" }, this.props.badgeValue)
+        );
+    }
+
+    private createBadgeLabel() {
+        return createElement("div",
+            {
+                className: classNames("widget-badge-display",
+                    { "widget-badge-link": !!this.props.badgeOnClick }
+                ),
+                onClick: () => this.handleOnClick(this.props.badgeOnClick)
+            },
+            DOM.span({ className: "widget-badge-text" }, this.props.label),
+            DOM.span({
+                className: classNames("widget-badge", "label",
+                    { [`label-${this.props.style}`]: !!this.props.style }
+                )
+            }, this.props.badgeValue)
+        );
+    }
+
+    private checkConfig() {
+        const errorMessage: string[] = [];
+        if (this.props.badgeOnClick.onClickType === "callMicroflow"
+            && !this.props.badgeOnClick.microflowProps.microflow) {
+            errorMessage.push("'On click' call a microFlow is set " +
+                "and there is no 'Microflow' Selected in tab Events");
+        }
+        if (this.props.badgeOnClick.onClickType === "showPage" && !this.props.badgeOnClick.pageProps.page) {
+            errorMessage.push("'On click' Show a page is set and there is no 'Page' Selected in tab 'Events'");
+        }
+        if (errorMessage.length > 0) {
+            errorMessage.unshift("Error in configuration of the Badge widget");
+            window.mx.ui.error(errorMessage.join("\n"));
```

Comment:
The (new) way of feedback is to provide inline and not in a popup. 
Only use popup when element could not be displayed, or horrible breaks the UI

Use the bootstrap alert.
```
import { DOM, StatelessComponent } from "react";

export const ValidationAlert: StatelessComponent<{ message: string }> = (props) =>
    DOM.div({ className: "alert alert-danger widget-validation-message" }, props.message);
```